webapp/tests/karma/unit/services/target-generator.js

```diff
@@ -40,6 +51,7 @@ describe('TargetGenerator service', function() {
   });
 
   it('returns empty array when no targets are configured', function(done) {
+    RulesEngine.enabled = false;
```

Comment:
I don't think this is right. The test description says "no targets are configured" but this is about the engine being disabled. I think it's now testing the same case as the test you added.